orderer/consensus/smartbft/util_network_test.go

```diff
@@ -74,6 +97,30 @@ func (ns *NetworkSetupInfo) CreateNodes(numberOfNodes int) map[uint64]*Node {
 	return nodeIdToNode
 }
 
+func (ns *NetworkSetupInfo) AddNewNode() (map[uint64]*Node, *Node) {
+	numberOfNodes := uint64(len(ns.nodeIdToNode))
+	numberOfNodes = numberOfNodes + 1
+	newNodeId := numberOfNodes
+
+	ns.t.Logf("Adding node %v to the network", newNodeId)
+
+	var ledgerToSyncWith []*cb.Block
+	for _, n := range ns.nodeIdToNode {
+		if n.NodeId != newNodeId {
+			ledgerToSyncWith = n.State.GetLedgerArray()
+		}
+	}
```

Comment:
why are you copying over and over from every node? you have to pick one existing node with the largest ledger.

Reply:
Right, a break command is missing here (the idea was to find some node and take its ledger), but pick the largest ledger is preferred.